torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -3155,170 +3155,177 @@ std::tuple<Tensor, Tensor> linalg_lstsq_backward(
 std::tuple<Tensor, Tensor> linalg_qr_jvp(
   const Tensor& dA,
   const Tensor& Q,
-  const Tensor& R
+  const Tensor& R,
+  const c10::string_view mode
 ) {
+  // dA = dQR + QdR
+  //
+  // Case m >= n
+  // We can put dQ in terms of dR
+  // dQ = dAR^{-1} - QdRR^{-1}
+  // Then we have
+  // Q^H dA R^{-1} = Q^HdQ + dRR^{-1}
+  // where Q^HdQ is skew Hermitian and dRR^{-1} is upper triangular
+  // Define sym(X) = X + X^H
+  // sym(dRR^{-1}) = sym(Q^H dA R^{-1})
+  // and define syminv(X) = triu(X) - 0.5 * diag(X) the inverse of sym : Triu(k) -> Her(k) to give
+  // dR = syminv(sym(Q^H dA R^{-1}))R
```

Comment:
I think this works because the diagonal of `R` is real?

Reply:
Indeed. It read: "the inverse of `sym : Triu(k, diag \in R) -> \Her(k)`"